Add credential_type to event in OTPFormAuthenticator and UsernamePasswordForm

jirutka opened the pull request:
Closes #40118

jirutka committed:
Add credential_type to event in OTPFormAuthenticator and UsernamePasswordForm

Closes #40118

Signed-off-by: Jakub Jirutka <jakub@jirutka.cz>

ahus1 commented:
Can you please update a test case to check that the details are present for the event? 

Maybe the LoginTest#loginSuccess can be amended. [Some of the tests](https://github.com/keycloak/keycloak/blob/0b30c872c8a5b6c224f75ef6e4e1c8be4005d6e3/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/LoginTest.java#L446) in that file already check for events. 

This test might be one where you can add a check for an event around OTP - or maybe even both, and then you don't need to update the first one. 

https://github.com/keycloak/keycloak/blob/c22f76867f595f0f7c327376a66a08dca01949e8/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/storage/UserStorageOTPTest.java#L129

Thanks!

ahus1 committed:
Adding the credential type by default where possible.
Also adding all successful credential types on the login event, not only the last one.

Signed-off-by: Alexander Schwartz <alexander.schwartz@ibm.com>

ahus1 commented:
Added some ideas on how to make this more generic to pick up all credentials used for successfully logging in.  This needs more work, especially tests.
